src/Models/Test/ModelInstantiatorTest.php

```diff
@@ -593,6 +669,68 @@ public function testDatabaseModelToDataObjectNestedObjectWithRelationsGetterExce
         }
     }
 
+    /**
+     * When given a property that is defined as nullable, `propertyAllowsNull` should
+     * return true
+     * @return void
+     */
+    public function testPropertyAllowsNullCorrectlyReturnsWhenPropertyIsNullable()
+    {
+        $reflection_class = new ReflectionClass(ModelInstantiatorTestObject::class);
+        $property_reflection = $reflection_class->getProperties();
+        $property = 'nullableFakeHasOneRelation'; // this is defined as being nullable
+
+        // Search to make sure the property exists in the data object or we'll get a false
+        // positive for this test.
+        $found = array_search($property, array_column($property_reflection, 'name'));
+        $result = $this->model_instantiator->propertyAllowsNull($property, $property_reflection);
+
+        $this->assertTrue($result);
+        $this->assertGreaterThan(0, $found);
```

Comment:
If the `$found` is necessary for this test to be accurate, it might make sense to move this assertion to right below  where it is defined. Then maybe add a failure message to the assertion that explains why the setup is incorrect instead of just having the comment. This could be applied to the test below as well.